### PR TITLE
fix(stale): wrongly commenting on issues and potentially pull requests

### LINKED ISSUE #148: This PR is stale comment for all Issues today!
We are receiving a “This PR is stable comment” in all Issues today! I guess started since the latest commits today! It is spamming us and there are around 200 issues open which are receiving comment. Please take a look

### PR DESCRIPTION
**Pull request by jusx**
### Context
Comments are being created for issues when only pull requests was configured to be receiving stale comments. It's also possible that when only issues are configured for stale, pull requests are receiving comments as well.

fixes #148 

### Changes
- fixed the stale validator to pass the correct parameters to the [search api](https://help.github.com/articles/searching-issues-and-pull-requests/#search-only-issues-or-pull-requests).
- upgraded jest and added convenience command to run jest --watch with the correct version of jest in dev dependency.
- harden the tests for v1config transform

Note: Some sanity check should be performed with manual testing before merging since there is possibility of "spam" if this is still a bug.

#### EDIT

Also found that comments were not created for stale PRs for configs that was PR stale configured. This has been fixed by returning the correct status in the stale validator.
- [x] Manually verified.

### CODE REVIEW

File: lib/validators/stale.js [hunk 1]
```diff
@@ -13,10 +13,14 @@ class Stale extends Validator {
 
   async validate (context, validationSettings) {
     let days = validationSettings.days || 20
-    let typeSetting = validationSettings.type || ['issues', 'pull_request']
+    let typeSetting = validationSettings.type || ['issue', 'pr']
```
Comment (shine2lay) [inline lib/validators/stale.js]: just double checking, why put both issues and pr as default. not a deal breaker but if it were me, i would rather have a less is more approach

Comment (jusx) [reply lib/validators/stale.js]: Because you wouldn't be specifying a `stale` validator if you don't intend for it to do something. 
Let's say the config has `stale` but no `type` specified it wouldn't do anything. Why make the user configure more when they just want to be notified of all stale issues and PR?

I would argue that in fact there might be more code if `type` is not specified.  You'd have to add to an error to the result. You'll have a constant and an if statement. 

Unless you want to just do a return when `type` is not specified -- that might be confusing to the user but -- definitely less code.

In short, whichever behavior we want the number of lines of code is not that much different.

File: __tests__/configuration/transformers/v1Config.test.js [hunk 2]
```diff
@@ -55,6 +55,60 @@ describe('Tests for description no empty scenarios transformations', () => {
...
+    loadWith()
+  })
+
+  test('Configuration in pulls only.', () => {
+    let whenSchedule = loadWith(`
+        stale:
+          days: 20
+          message: 'This is issue is stale. Please follow up!'
+    `, 1)
+    expect(whenSchedule[0].validate[0].type).toBe('pull_request')
```
[the remark this answers is not in the document]
Comment (jusx) [reply __tests__/configuration/transformers/v1Config.test.js]: Good call. I've added it.

File: __tests__/configuration/transformers/v1Config.test.js [hunk 3]
```diff
@@ -55,6 +55,60 @@ describe('Tests for description no empty scenarios transformations', () => {
...
+
+  test('Configuration in issues only', () => {
+    let whenSchedule = loadWith(`
+      issues:
+        stale:
+          days: 20
+          message: 'This is issue is stale. Please follow up!'
+    `, 1)
+
+    expect(whenSchedule[0].validate[0].type).toBe('issues')
```
[the remark this answers is not in the document]
Comment (jusx) [reply __tests__/configuration/transformers/v1Config.test.js]: Yup. Added.
